gallery/gallery/lib/demos/material/dialog_demo.dart

```diff
@@ -124,38 +124,48 @@ class DialogDemo extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    return Scaffold(
-      key: _scaffoldKey,
-      appBar: AppBar(
-        title: Text(_title(context)),
-      ),
-      body: Center(
-        child: RaisedButton(
-          child: Text(GalleryLocalizations.of(context).dialogShow),
-          onPressed: () {
-            switch (type) {
-              case DialogDemoType.alert:
-                _showAlertDialog(context);
-                break;
-              case DialogDemoType.alertTitle:
-                _showAlertDialogWithTitle(context);
-                break;
-              case DialogDemoType.simple:
-                _showSimpleDialog(context);
-                break;
-              case DialogDemoType.fullscreen:
-                Navigator.push<void>(
-                  context,
-                  MaterialPageRoute(
-                    builder: (context) => _FullScreenDialogDemo(),
-                    fullscreenDialog: true,
-                  ),
-                );
-                break;
-            }
-          },
-        ),
-      ),
+    return Navigator(
+      // Adding [ValueKey] to make sure that the widget gets rebuilt when
+      // changing type.
+      key: ValueKey(type),
+      onGenerateRoute: (settings) {
+        return MaterialPageRoute<Widget>(
```

Comment:
Is the widget returned by this route used anywhere? I can see some pop statements on 173 and 205, but those look like they're cutting through this navigator and going to the root insatnce.

Reply:
This is to ensure that the fullscreen dialog (`_FullScreenDialogDemo`) does not take up the entire screen and instead just the space within the Navigator that is defined here.

It is true that we have `rootNavigator: true` on those lines, not sure why and it does not seem to work exactly as expected. I'll have to take a second look at this after the holidays I think.

Reply:
The Navigator part I totally understand and support. I was just looking at the type parameter on line 132 and nitpickingly wondering if it should be `MaterialPageRoute<void>`. 😄

Reply:
Thanks for explaining further. That makes sense, changed it to void now because as you say whenever we use this Navigator it is never with any value. 😄 

Also I had to change the MaterialPageRoute to one without any transition animations (unrelated to this issue).